### DISCUSSION TITLE
Play CDN - support more settings

### DISCUSSION THREAD

**Discussion by Triloworld** [tailwindlabs/tailwindcss#6473] [Ideas]
As documentation state:
> Use the Play CDN to try Tailwind right in the browser without any build step. The Play CDN is designed for development purposes only, and is not the best choice for production.

For this it can use more settings and/or document how to connect with it.
We focus on built in plugins as example:
 ```
 plugins: [
    require("@tailwindcss/typography"),
    require("@tailwindcss/forms")({
      strategy: 'class',
    }),
    require("@tailwindcss/aspect-ratio"),
    require("@tailwindcss/line-clamp"),
  ],
```

I try to use settings in Browser Sync in "proxy" part to inject JS for development purpose.
This give unique option to work simultaneously on one page style 🤩  I change code and is injected with all setup when my friend edit in browser and we get page now ultra fast and have more iteration than before.

Plugins won't work as they aren't included in default CDN bundle.
If not then maybe add information that it isn't included and can be added after core one.

**Comment by bharathwaaj** [comment]
This should work

```
  <script src="https://cdn.tailwindcss.com?plugins=forms,typography,aspect-ratio,line-clamp"></script>
```

**Reply by Triloworld** [reply]
This isn't in documentation. That neat! :D

**Reply by vanling** [reply]
Do you still have to add them in the config?

**Reply by Triloworld** [reply]
No. I checked and nothing needed. They are added in default state. 
If you try (1 to 1 with require syntax) you get error like this one: 
> ?plugins=forms,typography,aspect-ratio,line-clamp:1 Uncaught (in promise) TypeError: Cannot read properties of null (reading '__isOptionsFunction')


Try this in injecting settings`tailwindConfig.plugins = [("@tailwindcss/forms")({
    strategy: 'class',
  })]`:
You then get error like this one:
> TypeError: "@tailwindcss/forms" is not a function

Maybe it isn't supported right now as bundled one only provide default setup.

**Reply by drew-smithies** [reply]
@Triloworld The require syntax works now. I did: 

`plugins: [ require('@tailwindcss/forms')({ strategy: 'class' }) ]`

and it worked great

**Comment by Triloworld** [comment]
Added to documentation: https://github.com/tailwindlabs/tailwindcss.com/pull/1060

**Comment by Triloworld** [comment]
Can you close discussion? Added PR :) @jenstornell

**Reply by vanling** [reply]
Nah, is nice to have it here if someone searches for it and missed it in the documentation <3

**Reply by Triloworld** [reply]
I added PR to documentation about It <3 Open Source rule :D But yes, for now is documentation. Thx @vanling for reply ! :)

**Reply by vanling** [reply]
No thank you! i just searched "CDN plugins" and found this post.

**Comment by dani-ooo** [comment]
how do you go about other functions, like `addVariant`?

**Reply by vanling** [reply]
Not sure,  are those variants not covered by JIT already?

**Reply by dani-ooo** [reply]
sorry, i’m new to TW. how does JIT work? could i add a `@supports` rule with it, like the `addVariant` [example in the documentation](https://tailwindcss.com/docs/plugins#adding-variants)?

**Reply by vanling** [reply]
@dani-ooo i am not sure exactly. have not used those addvariants before. 
Do you know if you are using the latest version v3.0.7? 

JIT should be enabled in latest version and then the following should apply. **but i could be wrong, because this is about the cdn version** 
> Every variant is enabled out of the box. Variants like focus-visible, active, disabled, and others are not normally enabled by default due to file-size considerations. Since this library generates styles on demand, you can use any variant you want, whenever you want. You can even stack them like sm:hover:active:disabled:opacity-75. Never configure your variants again.
from: https://tailwindcss.com/blog/just-in-time-the-next-generation-of-tailwind-css

**Reply by vanling** [reply]
more info about variants in docs for cdn usage: https://tailwindcss.com/docs/upgrade-guide#replace-variants-with-layer

**Reply by Triloworld** [reply]
`addVariant` - I don't see support from cdn to custom plugins sadly. CDN is for simple and fast approach. Is new concept, propably this topic showing it need to be extend. Maybe someone can provide API example how to use `addVariant` as custom plugin?

**Reply by sntran** [reply]
The `tailwind` global object has a `plugin` function that behaves the same as `tailwindcss/plugin`.

I have been using the following tailwind.config.js **file** that can be used with both the Play CDN and the CLI when needed:

```js
let tailwind;
try {
  tailwind = require("tailwindcss");
} catch (_) {
  tailwind = window.tailwind || {};
}

const plugin = tailwind.plugin;

tailwind.config = {
  theme: {
    ...
  },
  plugins: [
    plugin(function siblingVariants({ addVariant, e }) {
      // Adjacent sibling combinator
      addVariant("sibling", "& + *");
      // General sibling combinator
      addVariant("siblings", "& ~ *");
    }),
  ],
};

if (typeof module !== 'undefined' && module.exports) {
  module.exports = tailwind.config;
}
```

And on the HTML:

```html
<script src="https://cdn.tailwindcss.com?plugins=aspect-ratio,line-clamp"></script>
<script src="/css/tailwind.config.js"></script>
```

**Comment by Abdillah** [comment]
How to avoid tailwind `<style />` tag inserted earlier (not just before closing head)?
I have other style file that overrides Tailwind CSS.
